src/hooks/useRemainingRelays.ts

```diff
@@ -0,0 +1,54 @@
+import useAsync from '@/hooks/useAsync'
+import useSafeAddress from '@/hooks/useSafeAddress'
+import { Errors, logError } from '@/services/exceptions'
+import {
+  IS_PRODUCTION,
+  SAFE_GELATO_RELAY_SERVICE_URL_PRODUCTION,
+  SAFE_GELATO_RELAY_SERVICE_URL_STAGING,
+} from '@/config/constants'
+import { FEATURES, hasFeature } from '@/utils/chains'
+import { useCurrentChain } from '@/hooks/useChains'
+import { cgwDebugStorage } from '@/components/sidebar/DebugToggle'
+
+export const SAFE_GELATO_RELAY_SERVICE_URL =
+  IS_PRODUCTION || cgwDebugStorage.get()
+    ? SAFE_GELATO_RELAY_SERVICE_URL_PRODUCTION
+    : SAFE_GELATO_RELAY_SERVICE_URL_STAGING
+
+const fetchRemainingRelays = async (chainId: string, address: string): Promise<number> => {
+  const url = `${SAFE_GELATO_RELAY_SERVICE_URL}/${chainId}/${address}`
+
+  try {
+    const res = await fetch(url)
+    const data = await res.json()
+    return data.remaining
+  } catch (error) {
+    logError(Errors._630, (error as Error).message)
+    return 0
+  }
+}
+
+export const useRemainingRelaysBySafe = () => {
```

Comment:
Please always add return types to functions. Although it will likely infer it, it's easier for humans to read when a function's signature is typed.

Reply:
Return types speed up type checking as well.